backend/routes/userRoutes.js

```diff
@@ -0,0 +1,95 @@
+const express = require('express')
+const router = express.Router()
+const { PrismaClient } = require('@prisma/client')
+const prisma = new PrismaClient()
+const bcrypt = require('bcrypt');
+
+router.get('/', async (req, res) => {
+  const users = await prisma.user.findMany()
+  res.json(users)
+})
+
+router.post('/', async (req, res) => {
+  const {  username, email, password  } = req.body
+
+  const existingUser = await prisma.user.findFirst({ where: { username: username } });
+  if (existingUser) {
+    return res.status(400).json({ error: 'Username already exists' });
+  }
+
+  const hashedPassword = await bcrypt.hash(password, 10);
+  const hashedPasswordValue = String(hashedPassword);
+  
+  const newUser = await prisma.user.create({
+    data: {
+        username, 
+        email, 
+        password: hashedPasswordValue
+    }
+  })
+
+  req.session.user = newUser;
+
+  res.json(newUser)
+})
+
+router.put('/:id', async(req, res) => {
+  const { id } = req.params
+  const {  username, email, password  } = req.body
+
+  const updatedUser = await prisma.user.update({
```

Comment:
What should we expect if the user id doesn't exist in db? Like a malicious api call to your server?

Reply:
do we also need to check if the username is the same as the one logged in?

Reply:
Can I check it in the frontend?

Reply:
> What should we expect if the user id doesn't exist in db? Like a malicious api call to your server?

I think I'm going to delete all the actions related to user (except get and delete)

Reply:
I don't think I'm following your decision here in deleting all the actions. Can you elaborate a bit more?
I am suggesting for some sanity checks and error handling for update's use case here.

Reply:
Why not check username right here? You have username passed in to the req.body, and updatedUser.username retrieved from prisma table

Reply:
To Do:
check username in userRoutes